app/core/core-datastore-public/src/main/kotlin/com/hedvig/android/core/datastore/SettingsDataStore.kt

```diff
@@ -8,24 +8,22 @@ import com.hedvig.android.theme.Theme
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
 
-interface SettingsDataStore {
-  suspend fun setTheme(theme: Theme)
-  fun observeTheme(): Flow<Theme>
-}
-
-internal class SettingsDataStoreImpl(
+class SettingsDataStore(
   private val dataStore: DataStore<Preferences>,
-) : SettingsDataStore {
-
-  private val themeKey = stringPreferencesKey("settings-theme")
-
-  override suspend fun setTheme(theme: Theme) {
+) {
+  suspend fun setTheme(theme: Theme) {
     dataStore.edit {
       it[themeKey] = theme.name
     }
   }
 
-  override fun observeTheme(): Flow<Theme> {
-    return dataStore.data.map { it[themeKey]?.let { themeString -> Theme.valueOf(themeString) } ?: Theme.LIGHT }
+  fun observeTheme(): Flow<Theme> {
+    return dataStore.data.map {
+      it[themeKey]?.let { themeString -> Theme.valueOf(themeString) } ?: Theme.SYSTEM_DEFAULT
```

Comment:
Thought `Theme.SYSTEM_DEFAULT` would make more sense as a default, until we decide what happens on the first release of this. In any case, this is now not really hooked up with actually changing the app theme, selecting a new theme doesn't actually change the UI. So if we do in fact release without this being an option the better option would be to not even have this button in the first place. So lets keep the implementation here realistic, aka supporting dark mode, and if we do not have dark mode on for the release, let's just hide the button instead.